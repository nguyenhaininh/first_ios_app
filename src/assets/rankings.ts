import { ImageRequireSource } from 'react-native';

export interface RankingData {
  title: string;
  date: string;
  month: number;
  image: ImageRequireSource;
  description: string;
}

export default {
  overall: [
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-1.png'),
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-3.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-33.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-15.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-13.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-20.png')
    },
  ],
  lifestyle: [
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-1.png'),
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-3.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-33.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-15.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-13.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-20.png')
    },
  ],
  fashion: [
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-1.png'),
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-3.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-33.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-15.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-13.png')
    },
    {
      title: '雑誌タイトル',
      date: '2018.5.15',
      month: 1,
      description: 'デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの',
      image: require('./images/dummy/cover-20.png')
    },
  ],
};
