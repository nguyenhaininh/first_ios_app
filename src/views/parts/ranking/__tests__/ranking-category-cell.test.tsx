/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import * as React from 'react';
import RankingCategoryCell from '../ranking-category-cell';

import * as renderer from 'react-test-renderer';
import rankings from '../../../../assets/rankings';

it('renders correctly', () => {
  const fn = jest.fn();
  const tree = renderer.create(
    <RankingCategoryCell
      onPress={fn}
      data={rankings.overall[0]} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
