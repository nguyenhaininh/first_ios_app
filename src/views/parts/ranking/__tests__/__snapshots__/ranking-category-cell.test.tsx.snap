// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "height": 156,
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/images/dummy/cover-1.png",
      }
    }
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "height": 126,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      雑誌タイトル  [missing "en.ranking_month" translation]
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={3}
      style={
        Object {
          "color": "#333333",
          "fontSize": 12,
          "marginVertical": 10,
        }
      }
    >
      デザインのいい仕事場。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの。 WORK IS FUN! 仕事、はかどってますか？ いいアイデアが出ないのは、 ワーキングスペースの
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#666666",
            "fontSize": 11,
          }
        }
      >
        [missing "en.ranking_deliver" translation]
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#e62222",
            "borderWidth": 1,
            "height": 12,
            "justifyContent": "center",
            "marginLeft": 10,
            "width": 27,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#e62222",
              "fontSize": 8,
            }
          }
        >
          NEW
        </Text>
      </View>
    </View>
  </View>
</View>
`;
